perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/ElixirBonaFideStatus.java

```diff
@@ -0,0 +1,141 @@
+import cz.metacentrum.perun.registrar.model.ApplicationFormItemData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Registration module for ElixirBonaFideStatus.
+ * Contains logic if the user can acquire bonaFideStatus defined by ELIXIR.
+ *
+ * @author Dominik Frantisek Bucik <bucik@ics.muni.cz>
+ */
+public class ElixirBonaFideStatus implements RegistrarModule {
+
+	final static Logger log = LoggerFactory.getLogger(ElixirBonaFideStatus.class);
+
+	private static final String USER_BONA_FIDE_STATUS_ATTR_NAME = "bonaFideStatus";
+	private static final String USER_BONA_FIDE_STATUS_REMS_ATTR_NAME = "elixirBonaFideStatusREMS";
+	private static final String USER_AFFILIATIONS_ATTR_NAME = "eduPersonScopedAffiliations";
+	private static final String GROUP_ATESTATION_ATTR_NAME = "attestation";
+
+	private static final String A_U_D_userBonaFideStatus = AttributesManager.NS_USER_ATTR_DEF + ':' + USER_BONA_FIDE_STATUS_ATTR_NAME;
+	private static final String A_U_D_userBonaFideStatusRems = AttributesManager.NS_USER_ATTR_DEF + ':' + USER_BONA_FIDE_STATUS_REMS_ATTR_NAME;
+	private static final String A_U_D_userEduPersonScopedAffiliations = AttributesManager.NS_USER_ATTR_VIRT + ':' + USER_AFFILIATIONS_ATTR_NAME;
+	private static final String A_G_D_groupAttestation = AttributesManager.NS_GROUP_ATTR_DEF + ':' + GROUP_ATESTATION_ATTR_NAME;
+
+	private RegistrarManager registrar;
+
+
+	@Override
+	public void setRegistrar(RegistrarManager registrar) {
+		this.registrar = registrar;
+	}
+
+	@Override
+	public List<ApplicationFormItemData> createApplication(PerunSession user, Application application, List<ApplicationFormItemData> data) throws PerunException {
+		return data;
+	}
+
+	/**
+	 * Add new bonaFideStatus to the user attribute.
+	 */
+	@Override
+	public Application approveApplication(PerunSession session, Application app) throws PerunException {
+		User user = app.getUser();
+		Group group = app.getGroup();
+
+		AttributesManager am = session.getPerun().getAttributesManager();
+		Attribute attestation = am.getAttribute(session, group, A_G_D_groupAttestation);
+
+		String newValue = attestation.valueAsString();
+
+		Attribute bonaFideStatus = am.getAttribute(session, user, A_U_D_userBonaFideStatus);
+
+		List<String> value = new ArrayList<>();
+		if (bonaFideStatus.getValue() == null) {
+			value = bonaFideStatus.valueAsList();
+		}
+
+		value.add(newValue);
+
+		bonaFideStatus.setValue(value);
+		am.setAttribute(session, user, bonaFideStatus);
+
+		return app;
+	}
+
+	@Override
+	public Application rejectApplication(PerunSession session, Application app, String reason) throws PerunException {
+		return app;
+	}
+
+	@Override
+	public Application beforeApprove(PerunSession session, Application app) throws PerunException {
+		User user = app.getUser();
+		Group group = app.getGroup();
```

Comment:
Group is present only for group type applications. Are you sure, that nobody will use this registration module on VO ? I would prefer checking group on null before passing it to getAttribute() and not to rely on its check. Maybe skip all logic if group is not present (depends on your use-case). Right now it will throw ugly exception and user/admin will see only something like: "something went wrong - please report".